### PR TITLE
Add traditional Chinese (tw) language

### PR DESCRIPTION
**Pull request by hrs113355**
1. add locale `zh-TW` 
2. detect accept language from `Accept-Language` with region

### CODE REVIEW

**Comment by tw0517tw** [comment]
It would be great if we have zh-TW translation.

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -69,6 +69,6 @@ def user_locale
   end
 
   def http_head_locale
-    http_accept_language.compatible_language_from(I18n.available_locales)
+    http_accept_language.language_region_compatible_from(I18n.available_locales)
```
Comment (JuanitoFatas) [inline app/controllers/application_controller.rb]: why change this line?

File: config/locales/zh-TW.yml [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        sign_in: "登入"
+        sign_up: "註冊"
+        sign_out: "登出"
+      footer:
+        about: "關於"
+        blog: "部落格"
+        guides: "教學文件"
+        contribute: "貢獻"
+        help: "問題 / 回報"
+        status: "Status"
```
Comment (JuanitoFatas) [inline config/locales/zh-TW.yml]: How about 狀態?

File: config/locales/zh-TW.yml [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+        uptime: "Uptime"
```
Comment (JuanitoFatas) [inline config/locales/zh-TW.yml]: How about 運作時間? 上線時間?

File: config/locales/zh-TW.yml [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+        source_code: "原始碼"
+        data: "資料"
+        discussion_forum: "討論群組"
+        statistics: "統計"
+
+  pages:
+    about:
+      title: 關於
+      purpose:
+        header: "歡迎到 %{site}，這裡是 Ruby 社群 Gem 套件管理服務。此專案的建立有以下三個目的:"
```
Comment (JuanitoFatas) [inline config/locales/zh-TW.yml]: - 歡迎到 => 歡迎來到?
- should use chinese colon ： instead of English one

  `三個目的:`

File: config/locales/zh-TW.yml [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+      will_email_notice: "系統將會寄一封包含重設密碼連結的電子郵件給你"
+      submit: "更新密碼"
+
+  profiles:
+    edit:
+      title: "編輯個人檔案"
+      hide_email: "在公開的個人頁面中隱藏 email"
+      reset_password:
+        title: "重設密碼"
+        text: "需要重設密碼？ 沒問題。 %{link}"
```
Comment (JuanitoFatas) [inline config/locales/zh-TW.yml]: There is an extra space between `？` and `沒`.

File: config/locales/zh-TW.yml [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+        header: "相關連結"
+        home: "首頁"
+        code: "原始碼"
+        docs: "文件"
+        wiki: "Wiki"
+        mail: "郵件群組"
+        bugs: "Bug 追蹤"
+      versions_header: "版本列表"
+      bundler_header: Gemfile
+      show_all_versions: "顯示所有版本 (共 %{count})"
```
Comment (JuanitoFatas) [inline config/locales/zh-TW.yml]: Please use Chinese parentheses in `(共 %{count})`.

File: config/locales/zh-TW.yml [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+      licenses_header: License
+      no_licenses: N/A
+      requirements_header: "需求"
```
Comment (JuanitoFatas) [inline config/locales/zh-TW.yml]: How about `必填`?

File: config/locales/zh-TW.yml [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+
+  searches:
+    show:
+      subtitle: "%{query}"
+      exact_match: "完全符合"
+
+  sessions:
+    new:
+      forgot_password: "忘記密碼?"
```
Comment (JuanitoFatas) [inline config/locales/zh-TW.yml]: Please use Chinese question mark.

File: config/locales/zh-TW.yml [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+
+  pages:
+    about:
+      title: 關於
+      purpose:
+        header: "歡迎到 %{site}，這裡是 Ruby 社群 Gem 套件管理服務。此專案的建立有以下三個目的:"
+        better_api: "提供更好的 Gem API"
+        transparent_pages: "建立更透明、更易於瀏覽的 Gem 專案頁面"
+        enable_community: "讓社群能透過眾人的力量來改善本站"
+      founding: "這個專案是由 %{founder} 於 2009 年 4 月建立，發展過程中有超過 %{contributors} 貢獻者加入以及 %{downloads}。 而自從 RubyGems 1.3.6 發布之後，本站的名稱由 Gemcutter 改名為 %{title}，表示本站在 Ruby 社群中的核心角色。"
```
Comment (JuanitoFatas) [inline config/locales/zh-TW.yml]: I would translate to this:

```
本專案由 %{founder} 於 2009 年 4 月創立，發展過程中有超過 %{contributors} 貢獻者以及 %{downloads}。自 RubyGems 1.3.6 發佈以來，本站名稱由 Gemcutter 更名為 %{title}，本站自此之後成為 Ruby 社群的核心網站。
```

Could you see if any could be incorporated into yours from my above translation?

File: config/locales/zh-TW.yml [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+        dashboard: "控制台"
+        sign_in: "登入"
+        sign_up: "註冊"
+        sign_out: "登出"
+      footer:
+        about: "關於"
+        blog: "部落格"
+        guides: "教學文件"
+        contribute: "貢獻"
+        help: "問題 / 回報"
```
Comment (chitsaou) [inline config/locales/zh-TW.yml]: Any good reason not to use a single phrase `說明`?

File: config/locales/zh-TW.yml [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+    about:
+      title: 關於
+      purpose:
+        header: "歡迎到 %{site}，這裡是 Ruby 社群 Gem 套件管理服務。此專案的建立有以下三個目的:"
+        better_api: "提供更好的 Gem API"
+        transparent_pages: "建立更透明、更易於瀏覽的 Gem 專案頁面"
+        enable_community: "讓社群能透過眾人的力量來改善本站"
+      founding: "這個專案是由 %{founder} 於 2009 年 4 月建立，發展過程中有超過 %{contributors} 貢獻者加入以及 %{downloads}。 而自從 RubyGems 1.3.6 發布之後，本站的名稱由 Gemcutter 改名為 %{title}，表示本站在 Ruby 社群中的核心角色。"
+      support: "雖然 Gemcutter 並不是由一個特定的公司運作，但在發展過程中接受了許多來源的幫助。目前的設計、圖像以及網站的前端開發是由 %{dockyard} 提供。%{github} 也幫助我們能更容易地協作和分享原始碼。本站部署在 %{heroku} 上，其一流的服務，更有助於證明 Gemcutter 是一個可以穩定、可行的解決方案。"
+      technical: "關於本站的技術資訊：100% Ruby。主站是一個 %{rails} 應用程式，RubyGem 則是透過 %{sinatra} 提供服務。Gems 架設在 %{s3} 上，使得 Gem 從發佈到提供下載的時間大幅縮短。詳細資訊可從 GitHub 上的 %{source_code} 查看 (遵守 %{license} 協議)"
```
Comment (JuanitoFatas) [inline config/locales/zh-TW.yml]: Please use Chinese parens.

File: config/locales/zh-TW.yml [hunk 12]
```diff
@@ -0,0 +1,163 @@
...
+  footer_about: "RubyGems.org 是 Ruby 社群的 Gem 套件管理服務，讓你能立即地發佈及安裝你的 Gem 套件，並且利用 API 查詢及操作可用 Gem 的詳細資訊。<br />分享你的貢獻，讓 RubyGems.org 變得更加完善。"
+
+  dashboards:
+    show:
+      latest: "最近更新"
+      latest_title: "最新的 RSS Feed"
+      no_subscriptions: "你還沒有訂閱過 Gem，前往 %{gem_link} 來訂閱！"
+      gem_link_text: "Gem 頁面"
+      mine: "我的 Gems"
+      no_owned: "你還沒有發佈過 Gem，請參考%{creating_link}教學，或從 RubyForge 的%{migrating_link}教學。"
```
Comment (JuanitoFatas) [inline config/locales/zh-TW.yml]: How about?

```
你尚未發佈任何 Gem。可以閱讀 %{creating_link} 教學，或參考 %{migrating_link} 教學來將你的 Gem 從 RubyForge 遷移過來。
```

File: config/locales/zh-TW.yml [hunk 13]
```diff
@@ -0,0 +1,163 @@
...
+  please_sign_up: "無法存取，請先在 https://rubygems.org 上註冊帳號"
+  feed_subscribed: "RubyGems.org | 訂閱 Gems"
+  feed_latest: "RubyGems.org | 最新 Gems"
+  time_ago: "%{duration} 前"
+  form_disable_with: "請稍候..."
+  update: "更新"
+  edit: "編輯"
+  sign_in: "登入"
+  sign_up: "註冊"
+  footer_about: "RubyGems.org 是 Ruby 社群的 Gem 套件管理服務，讓你能立即地發佈及安裝你的 Gem 套件，並且利用 API 查詢及操作可用 Gem 的詳細資訊。<br />分享你的貢獻，讓 RubyGems.org 變得更加完善。"
```
Comment (JuanitoFatas) [inline config/locales/zh-TW.yml]: For this part:

```
分享你的貢獻，讓 RubyGems.org 變得更加完善。
```

How about?

```
現在就成為貢獻者，貢獻一己之力來改善本站
```

File: config/locales/zh-TW.yml [hunk 14]
```diff
@@ -0,0 +1,163 @@
...
+        title: "重設密碼"
+        text: "需要重設密碼？ 沒問題。 %{link}"
+        link_text: "從這裡設定新密碼"
+      api_access:
+        title: API 存取
+        key_is: "你的 API key 是 %{key}。"
+        credentials: "如果你需要從 command line 中執行 %{gem_commands_link}，你需要先準備 %{gem_credentials_file} 這個檔案，用以下的指令可以產生："
+        link_text: "Gem 指令"
+        reset: "重設 API key"
+        confirm_reset: "確定要重設嗎？ 此動作無法還原"
```
Comment (JuanitoFatas) [inline config/locales/zh-TW.yml]: Extra space between `？` and `此` :mag_right:

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (hrs113355) [inline app/controllers/application_controller.rb]: `compatible_language_from` would ignore language region from `Accept-Language` and choose the first compatible language. For example:

```
  http_accept_language.user_preferred_languages # => %w(zh-TW)
  available = %w(zh-CN zh-TW)
  http_accept_language.compatible_language_from(available) # => 'zh-CN'

  http_accept_language.language_region_compatible_from(available) # => 'zh-TW'
```

Comment (JuanitoFatas) [inline app/controllers/application_controller.rb]: :+1: from me.

/cc @huacnlee

**Comment by JuanitoFatas** [comment]
LGTM :clap:

**Comment by sferik** [comment]
:cn: :exclamation:

**Comment by sferik** [comment]
Oops…wrong flag. Sorry about that. :disappointed:

Anyway, thank you for your contribution.

**Comment by JuanitoFatas** [comment]
:+1:

**Comment by hrs113355** [comment]
@sferik: Nevermind! Thank you for merging this!
@JuanitoFatas, @chitsaou: Many thanks for helping in translation!

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (huacnlee) [inline app/controllers/application_controller.rb]: :+1:

**Pull request merged**